tests/helper_test.php

```diff
@@ -0,0 +1,99 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace mod_kanban;
+
+/**
+ * Unit tests for helper class.
+ *
+ * @package    mod_kanban
+ * @category   test
+ * @copyright  2023-2025 ISB Bayern
+ * @author     Stefan Hanauska <stefan.hanauska@csg-in.de>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @covers     \mod_kanban\helper
+ */
+final class helper_test extends \advanced_testcase {
+    /**
+     * Tests the sequence_add_after method.
+     *
+     * @return void
+     */
+    public function test_sequence_add_after(): void {
+        // Test adding a new element after an existing element.
+        $this->assertEquals('1,2,3', helper::sequence_add_after('1,2', 2, 3));
+        $this->assertEquals('1,3,2', helper::sequence_add_after('1,2', 1, 3));
+        // Test adding a new element at the beginning.
+        $this->assertEquals('3,1,2', helper::sequence_add_after('1,2', 0, 3));
+        // Test adding a new element after a non existing element (should be added at the end).
+        $this->assertEquals('1,2,3', helper::sequence_add_after('1,2', 4, 3));
+        // Test adding a new element to an empty sequence.
+        $this->assertEquals('3', helper::sequence_add_after('', 0, 3));
+    }
+
+    /**
+     * Tests the sequence_remove method.
+     *
+     * @return void
+     */
+    public function test_sequence_remove(): void {
+        // Test removing an element.
+        $this->assertEquals('1,3', helper::sequence_remove('1,2,3', 2));
+        // Test removing the first element.
+        $this->assertEquals('2,3', helper::sequence_remove('1,2,3', 1));
+        // Test removing the last element.
+        $this->assertEquals('1,2', helper::sequence_remove('1,2,3', 3));
+        // Test removing a non existing element.
+        $this->assertEquals('1,2,3', helper::sequence_remove('1,2,3', 4));
+        // Test removing from an empty sequence.
+        $this->assertEquals('', helper::sequence_remove('', 1));
+    }
+
+    /**
+     * Tests the sequence_move_after method.
+     *
+     * @return void
+     */
+    public function test_sequence_move_after(): void {
+        // Test moving the last element after another element.
+        $this->assertEquals('1,3,2', helper::sequence_move_after('1,2,3', 1, 3));
+        // Test moving an element at the beginning.
+        $this->assertEquals('3,1,2', helper::sequence_move_after('1,2,3', 0, 3));
+        // Test moving an element after itself (should not change the sequence).
+        $this->assertEquals('1,2,3', helper::sequence_move_after('1,2,3', 2, 2));
+        // Test moving an element after a non existing element (should not change the sequence).
+        $this->assertEquals('1,2,3', helper::sequence_move_after('1,2,3', 4, 3));
+    }
```

Comment:
ggf. Test moving in an empty sequence?

Reply:
Ist hinzugefügt. Gleichzeitig habe ich noch überdacht, welches Verhalten denn sinnvoller ist, wenn das $afteritem (aus welchen Gründen auch immer) nicht existiert. Günstiger ist in meinen Augen, das Element dann am Ende einzufügen (um nicht zugeordnete Spalten / Karten in der Datenbank zu vermeiden).